### ISSUE TITLE
pyQT M1 Mac woes

### ISSUE THREAD

**Issue by johnaulich92** [DigiScore/neoscore#49]
I saw the Rosetta solution but it hasn't worked for me unfortunately:
Attempting to install PyQt5 directly generates a similar error.

```
johnaulich@john-2 ~ % python3.10 -m venv env   
johnaulich@john-2 ~ % source env/bin/activate  
(env) johnaulich@john-2 ~ % python --version
Python 3.10.4
(env) johnaulich@john-2 ~ % pip install --upgrade pip
Requirement already satisfied: pip in ./env/lib/python3.10/site-packages (22.1.2)
(env) johnaulich@john-2 ~ % pip install neoscore     
Collecting neoscore
  Using cached neoscore-0.1.1-py3-none-any.whl (934 kB)
Collecting img2pdf==0.4.3
  Using cached img2pdf-0.4.3.tar.gz (98 kB)
  Preparing metadata (setup.py) ... done
Collecting sortedcontainers==2.4.0
  Using cached sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Collecting PyQt5<6.0.0,>=5.15.6
  Using cached PyQt5-5.15.7.tar.gz (3.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      Querying qmake about your Qt installation...
      Traceback (most recent call last):
        File "/Users/johnaulich/env/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 156, in prepare_metadata_for_build_wheel
          hook = backend.prepare_metadata_for_build_wheel
      AttributeError: module 'sipbuild.api' has no attribute 'prepare_metadata_for_build_wheel'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/johnaulich/env/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/Users/johnaulich/env/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/johnaulich/env/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 160, in prepare_metadata_for_build_wheel
          whl_basename = backend.build_wheel(metadata_directory, config_settings)
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-h2vrtrt6/overlay/lib/python3.10/site-packages/sipbuild/api.py", line 46, in build_wheel
          project = AbstractProject.bootstrap('wheel',
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-h2vrtrt6/overlay/lib/python3.10/site-packages/sipbuild/abstract_project.py", line 87, in bootstrap
          project.setup(pyproject, tool, tool_description)
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-h2vrtrt6/overlay/lib/python3.10/site-packages/sipbuild/project.py", line 600, in setup
          self.update(tool)
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-install-gwb86hec/pyqt5_07517a0f769a496c8bef07a27bd9041a/project.py", line 166, in update
          raise UserException(
      sipbuild.exceptions.UserException
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

I'm on Mac OS X 12.4 and this is an M1 Max machine fwiw.

**Comment by ajyoon** [comment]
Out of curiosity, if you have other python 3.x versions available on your system, are you able to install PyQt5 in them? Maybe the wheels aren't as available for 3.10 since it's newer.

**Comment by ajyoon** [comment]
Does it help if you install 5.15.6 specifically with `pip install PyQt5==5.15.6`? Maybe 5.15.7 is broken

**Comment by johnaulich92** [comment]
Unfortunately no :( I have an intel machine I can dig out

**Comment by ajyoon** [comment]
sorry for the stalled progress here, we're going to be looking into lowering our minimum python version to 3.9 on a hunch it could avoid a lot of these issues. I think wheel support is just not great for newer python versions.

**Comment by johnaulich92** [comment]
Thanks Andrew, please keep me posted on this!

I’m happy to test anything out that you need me to.

**Comment by ajyoon** [comment]
Oh actually it would be very helpful if you could try installing PyQt5 on a Python 3.9 virtual environment. That would help us know if it would improve the situation.

**Comment by johnaulich92** [comment]
Unfortunately it's a similar problem

```johnaulich@john-2 ~ % python3.9 -m venv env
johnaulich@john-2 ~ % source env/bin/activate
(env) johnaulich@john-2 ~ % pip install --upgrade pip
Requirement already satisfied: pip in ./env/lib/python3.9/site-packages (22.1.2)
(env) johnaulich@john-2 ~ % pip install PyQt5        
Collecting PyQt5
  Using cached PyQt5-5.15.7.tar.gz (3.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [23 lines of output]
      Querying qmake about your Qt installation...
      Traceback (most recent call last):
        File "/Users/johnaulich/env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 156, in prepare_metadata_for_build_wheel
          hook = backend.prepare_metadata_for_build_wheel
      AttributeError: module 'sipbuild.api' has no attribute 'prepare_metadata_for_build_wheel'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/johnaulich/env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/Users/johnaulich/env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/johnaulich/env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 160, in prepare_metadata_for_build_wheel
          whl_basename = backend.build_wheel(metadata_directory, config_settings)
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-hugt05x0/overlay/lib/python3.9/site-packages/sipbuild/api.py", line 46, in build_wheel
          project = AbstractProject.bootstrap('wheel',
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-hugt05x0/overlay/lib/python3.9/site-packages/sipbuild/abstract_project.py", line 87, in bootstrap
          project.setup(pyproject, tool, tool_description)
        File "/private/var/folders/80/wdttptns7gz44ttky392mh4w0000gn/T/pip-build-env-hugt05x0/overlay/lib/python3.9/site-packages/sipbuild/project.py", line 600, in setup
          self.update(tool)
        File "project.py", line 166, in update
      sipbuild.exceptions.UserException
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.```

**Comment by ajyoon** [comment]
ah bummer.. does that work under rosetta by any chance?

Actually, I'm seeing now that [the prebuilt wheels only exist for 3.7](https://pypi.org/project/PyQt5/5.15.7/#files), could you try installing PyQt5 on 3.7?

**Comment by johnaulich92** [comment]
Okay, this took a bit of doing! PyQt5 successfully installs on 3.7 on Rosetta. 

Just fyi 3.7 itself won’t install natively.

**Comment by ajyoon** [comment]
Thanks for checking that! [We've just released neoscore 0.1.3](https://github.com/DigiScore/neoscore/discussions/56) which reduces the minimum supported Python version to 3.7, so you should be able to install neoscore on it now (with Rosetta). Please let us know if this works!

**Comment by johnaulich92** [comment]
Success! Thanks.

Now to actually try it out…
